tasks/nginx.yml

```diff
@@ -1,4 +1,9 @@
 ---
+- name: Set nginx max upload size
+  # Required to allow uploading firmware images
+  set_fact:
+    openwisp2_nginx_client_max_body_size: "{{openwisp2_firmware_upgrader_max_file_size}}"
+  when: openwisp2_firmware_upgrader
```

Comment:
I would add `and not openwisp2_nginx_client_max_body_size` so that we may be able to set a higher size to `openwisp2_nginx_client_max_body_size` if ever needed (imagine a different module which requires even bigger upload sizes). What do you think?

Reply:
`openwisp2_nginx_client_max_body_size` defaults to `20M`. We will have to change few other things to make it work. But, it is doable.

Reply:
ah ok sorry I forgot, no problem, let's keep it simple then

Reply:
here I have an idea, the condition could be:

```
when: openwisp2_firmware_upgrader and openwisp2_firmware_upgrader > openwisp2_nginx_client_max_body_size
```

What do you think?

Reply:
`openwisp2_firmware_upgrader_max_file_size` is bytes in `int`. `openwisp2_nginx_client_max_body_size` can be that, or in format like `10K`/``20M`. It won't work in all cases.